HotFix/Core/FIXMessageWriter.cs

```diff
@@ -8,123 +8,159 @@ public class FIXMessageWriter
         private const byte SOH = (byte)'\u0001';
         private const byte EQL = (byte) '=';
 
-        internal readonly byte[] _buffer;
-        internal int _end;
-
-        private readonly int _headerEnd;
+        private readonly byte[] _body;
         private int _bodyEnd;
-        private int _trailerEnd;
 
+        internal readonly byte[] Buffer;
+        internal int End;
 
-        public FIXMessageWriter(int maxLength, string beginString)
+        public FIXMessageWriter(int maxLength)
         {
-            _buffer = new byte[maxLength];
-
-            _end += _buffer.WriteString(_end, "8=");
-            _end += _buffer.WriteString(_end, beginString);
-            _buffer[_end++] = SOH;
-
-            _end += _buffer.WriteString(_end, "9=");
-            _end += _buffer.WriteString(_end, "00000");
-            _buffer[_end] = SOH;
-
-            _headerEnd = _end;
-            _end++;
-
-            _end += _buffer.WriteString(_end, "35=");
+            Buffer = new byte[maxLength];
+            _body = new byte[maxLength];
         }
 
-        public void Prepare(string messageType)
+        public FIXMessageWriter Clear()
         {
-            _end = _headerEnd + 4;
+            End = 0;
+            _bodyEnd = 0;
 
-            _end += _buffer.WriteString(_end, messageType);
-            _buffer[_end++] = SOH;
+            return this;
         }
 
-        public void Set(int tag, string value)
+        public FIXMessageWriter Prepare(string beginString, string messageType, long seqNum, DateTime sendingTime, string sender, string target)
         {
-            _end += _buffer.WriteInt(_end, tag);
-            _buffer[_end++] = EQL;
+            // write beginstring
+            End = Buffer.WriteString(0, "8=");
+            End += Buffer.WriteString(End, beginString);
+            Buffer[End++] = SOH;
+
+            // write length placeholder
+            // Should probably measure how many digits are needed to represent maxLength and use that many zeros
+            // The zeros string could then be cached and reused since it can be calculated at construction time
+            End += Buffer.WriteString(End, "9=00000");
+            var lengthPosition = End - 1; // points to the last zero
+            Buffer[End++] = SOH;
+
+            // write messagetype
+            End += Buffer.WriteString(End, "35=");
+            End += Buffer.WriteString(End, messageType);
+            Buffer[End++] = SOH;
```

Comment:
I like fast. I would have inlined this method into the kernel if i could. Also as discussed, we'll add that in a separate pull whip along with a "read character" method, benchmarks, tests et al.